oxapy/__init__.py

```
@@ -8,7 +11,143 @@
+        Sets the file patterns to monitor for changes.
+
+        Args:
+            p (list[str]): A list of glob patterns (e.g., ["*.py", "*.json"]).
+
+        Returns:
+            Oxapy: The current instance for method chaining.
+        """
+        self.__patterns = p
+        return self
+
+    def set_watch_dir(self, dir: str):
```

Ruff (0.15.21)

[error] Function argument `dir` is shadowing a Python builtin

(A002)